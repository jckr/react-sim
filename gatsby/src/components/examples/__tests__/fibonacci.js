import React from 'react';
import renderer from 'react-test-renderer';
import {
  initData,
  updateData,
  draw,
  FibonacciSquaresFrame,
} from '../fibonacci';
import ctx from '../../helpers/mock-ctx';

describe('Fibonacci', () => {
  const params = { size: 500 };
  let data;
  it('data function works', () => {
    data = initData(params);
    expect(data).toMatchSnapshot();
    for (let tick = 1; tick < 10; tick++) {
      data = updateData({ data, tick, params });
      expect(data).toMatchSnapshot();
    }
  });
  it('model renders properly', () => {
    expect(draw({ ctx, params, tick: 5 })).toMatchSnapshot();
    const squares = renderer
      .create(<FibonacciSquaresFrame data={data} tick={5} params={params} />)
      .toJSON();
    expect(squares).toMatchSnapshot();
  });
});
