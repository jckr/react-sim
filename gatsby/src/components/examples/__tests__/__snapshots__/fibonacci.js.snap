// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fibonacci data function works 1`] = `
Array [
  0,
]
`;

exports[`Fibonacci data function works 2`] = `
Array [
  0,
  1,
]
`;

exports[`Fibonacci data function works 3`] = `
Array [
  0,
  1,
  1,
]
`;

exports[`Fibonacci data function works 4`] = `
Array [
  0,
  1,
  1,
  2,
]
`;

exports[`Fibonacci data function works 5`] = `
Array [
  0,
  1,
  1,
  2,
  3,
]
`;

exports[`Fibonacci data function works 6`] = `
Array [
  0,
  1,
  1,
  2,
  3,
  5,
]
`;

exports[`Fibonacci data function works 7`] = `
Array [
  0,
  1,
  1,
  2,
  3,
  5,
  8,
]
`;

exports[`Fibonacci data function works 8`] = `
Array [
  0,
  1,
  1,
  2,
  3,
  5,
  8,
  13,
]
`;

exports[`Fibonacci data function works 9`] = `
Array [
  0,
  1,
  1,
  2,
  3,
  5,
  8,
  13,
  21,
]
`;

exports[`Fibonacci data function works 10`] = `
Array [
  0,
  1,
  1,
  2,
  3,
  5,
  8,
  13,
  21,
  34,
]
`;

exports[`Fibonacci model renders properly 1`] = `undefined`;

exports[`Fibonacci model renders properly 2`] = `
<div
  style={
    Object {
      "height": 500,
      "overflow": "hidden",
      "pointerEvents": "none",
      "position": "relative",
      "width": 500,
    }
  }
>
  <div
    style={
      Object {
        "height": "550px",
        "position": "relative",
        "transform": "scale(0.9090909090909091) translate(240px,400px)",
        "transformOrigin": "top left",
        "transition": "transform 0.2s",
        "width": "340px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#b6b6e2",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 10,
          "justifyContent": "center",
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": -10,
          "width": 10,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 6.666666666666667,
          }
        }
      >
        1
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 10,
            "boxSizing": "border-box",
            "height": 20,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": undefined,
            "width": 20,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#a6d3d9",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 10,
          "justifyContent": "center",
          "left": 10,
          "overflow": "hidden",
          "position": "absolute",
          "top": -10,
          "width": 10,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 6.666666666666667,
          }
        }
      >
        1
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 10,
            "boxSizing": "border-box",
            "height": 20,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(-50%)",
            "width": 20,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#ffeb3b",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 20,
          "justifyContent": "center",
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "width": 20,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 13.333333333333334,
          }
        }
      >
        2
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 20,
            "boxSizing": "border-box",
            "height": 40,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(-50%,-50%)",
            "width": 40,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#95d392",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 30,
          "justifyContent": "center",
          "left": -30,
          "overflow": "hidden",
          "position": "absolute",
          "top": -10,
          "width": 30,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        3
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 30,
            "boxSizing": "border-box",
            "height": 60,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(0,-50%)",
            "width": 60,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#b6b6e2",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 50,
          "justifyContent": "center",
          "left": -30,
          "overflow": "hidden",
          "position": "absolute",
          "top": -60,
          "width": 50,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 33.333333333333336,
          }
        }
      >
        5
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 50,
            "boxSizing": "border-box",
            "height": 100,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": undefined,
            "width": 100,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#a6d3d9",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 80,
          "justifyContent": "center",
          "left": 20,
          "overflow": "hidden",
          "position": "absolute",
          "top": -60,
          "width": 80,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 53.333333333333336,
          }
        }
      >
        8
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 80,
            "boxSizing": "border-box",
            "height": 160,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(-50%)",
            "width": 160,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#ffeb3b",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 130,
          "justifyContent": "center",
          "left": -30,
          "overflow": "hidden",
          "position": "absolute",
          "top": 20,
          "width": 130,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 43.333333333333336,
          }
        }
      >
        13
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 130,
            "boxSizing": "border-box",
            "height": 260,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(-50%,-50%)",
            "width": 260,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#95d392",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 210,
          "justifyContent": "center",
          "left": -240,
          "overflow": "hidden",
          "position": "absolute",
          "top": -60,
          "width": 210,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 70,
          }
        }
      >
        21
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 210,
            "boxSizing": "border-box",
            "height": 420,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": "translate(0,-50%)",
            "width": 420,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "#b6b6e2",
          "border": "#{borderWidth}px solid #777",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 340,
          "justifyContent": "center",
          "left": -240,
          "overflow": "hidden",
          "position": "absolute",
          "top": -400,
          "width": 340,
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 113.33333333333333,
          }
        }
      >
        34
      </div>
      <div
        style={
          Object {
            "border": "0.1px solid black",
            "borderRadius": 340,
            "boxSizing": "border-box",
            "height": 680,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "transform": undefined,
            "width": 680,
          }
        }
      />
    </div>
  </div>
</div>
`;
